src/main/java/org/mastodon/mamut/clustering/ui/DendrogramPanel.java

```diff
@@ -0,0 +1,345 @@
+package org.mastodon.mamut.clustering.ui;
+
+import com.apporiented.algorithm.clustering.Cluster;
+import com.apporiented.algorithm.clustering.visualization.ClusterComponent;
+import com.apporiented.algorithm.clustering.visualization.VCoord;
+import org.apache.commons.lang3.tuple.Pair;
+import org.mastodon.mamut.clustering.util.Classification;
+
+import javax.swing.JPanel;
+import java.awt.BasicStroke;
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.Stroke;
+import java.awt.geom.Line2D;
+import java.awt.geom.Rectangle2D;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Class for painting dendrograms derived from a {@link Classification} object.<p>
+ * In addition to {@link com.apporiented.algorithm.clustering.visualization.DendrogramPanel}, this class adds the possibility to paint an extra line on top of dendrogram (e.g. the cut-off line) and colorizing the clusters below this cut-off line.
+ *
+ * @author Stefan Hahmann
```

Comment:
I would not call it a copy anymore, since the changes compared to the original class a quite substantial. I called it a "re-implementation now". I added the differences to the original class more carefully now. Cf.: https://github.com/mastodon-sc/mastodon-deep-lineage/commit/ed1c892afe79f9000399950b993c7639e23b4389